diff --git a/lightning-rapid-gossip-sync/Cargo.toml b/lightning-rapid-gossip-sync/Cargo.toml
index 03efbde3..fb01aba8 100644
--- a/lightning-rapid-gossip-sync/Cargo.toml
+++ b/lightning-rapid-gossip-sync/Cargo.toml
@@ -12,6 +12,13 @@ Utility to process gossip routing data from Rapid Gossip Sync Server.
 [features]
 default = ["std"]
 std = ["bitcoin-io/std", "bitcoin_hashes/std"]
+max_level_off = []
+max_level_error = []
+max_level_warn = []
+max_level_info = []
+max_level_debug = []
+max_level_trace = []
+
 
 [dependencies]
 lightning = { version = "0.0.124", path = "../lightning", default-features = false }
